pypeit/specobj.py

```diff
@@ -629,4 +629,50 @@ def __repr__(self):
                 repr += '{}: {}\n'.format(key, rdict[key])
         return repr + '>'
 
+    def has_opt_ext(self):
+        """
+        Cehck that all the values of the optimal extraction exist
+
+        Returns:
+            :obj:bool: True if all OPT values are available
+        """
+        keys_to_check = ['OPT_WAVE', 'OPT_COUNTS', 'OPT_COUNTS_IVAR', 'OPT_MASK']
+        has_opt = []
+        for key in keys_to_check:
+            has_opt.append(True if self[key] is not None else False)
+        return np.all(has_opt)
```

Comment:
I doubt it matters much, but my guess is that list compression is faster here:
```
return np.all([self[key] is not None for key in keys_to_check])
```

Reply:
Thanks!